server-spi-private/src/main/java/org/keycloak/broker/provider/util/IdentityBrokerState.java

```diff
@@ -51,8 +51,8 @@ public static IdentityBrokerState decoded(String state, String clientId, String
                 bb.putLong(clientDbUuid.getMostSignificantBits());
                 bb.putLong(clientDbUuid.getLeastSignificantBits());
                 byte[] clientUuidBytes = bb.array();
-                clientIdEncoded = Base64.getEncoder().encodeToString(clientUuidBytes).replace("=", "");
-            } catch (IllegalArgumentException e) {
+                clientIdEncoded = Base64Url.encode(clientUuidBytes);
```

Comment:
Istn't the `.replace("=", "");`missing here?

Reply:
no, as the the method already take the part before the first "="
(see comment)

    public static String encodeBase64ToBase64Url(String base64) {
        String s = base64.split("=")[0]; // Remove any trailing '='s
        s = s.replace('+', '-'); // 62nd char of encoding
        s = s.replace('/', '_'); // 63rd char of encoding
        return s;
    }

Reply:
You are right, I haven't checked that. I just compared it to https://github.com/keycloak/keycloak/pull/15735/files but apparently standard Java does not remove trailing `=`?

Reply:
true, standard java doesn't, but since we know the boundaries from the base64, we don't need to keep the `=` for padding

Java 17 always use padding by default :
![padding](https://user-images.githubusercontent.com/45632394/205605626-22c260d7-53e7-4717-94e5-024e74d60bad.png)

but provide a way to disable it 

![java_no_padding](https://user-images.githubusercontent.com/45632394/205606223-7e8b82fe-f08d-436c-a133-1b27f04dc37e.png)